common/src/main/java/com/redhat/cloud/notifications/models/Event.java

```diff
@@ -94,6 +95,25 @@ public Event(String accountId, String orgId, EventType eventType, UUID eventId)
         eventTypeDisplayName = eventType.getDisplayName();
     }
 
+    /**
+     * Constructor used to build events that will be exported. The idea is that
+     * the exported events should look the same as they look in the event log.
+     * That is why just a few of the fields are used to construct an event,
+     * which are the fields that will be pulled from the database.
+     * @param id the {@link UUID} of the event.
+     * @param bundleDisplayName the display name of the bundle.
+     * @param applicationDisplayName the display name of the application.
+     * @param eventTypeDisplayName the display name of the event type.
+     * @param created the date in which the event was created.
+     */
+    public Event(final UUID id, final String bundleDisplayName, final String applicationDisplayName, final String eventTypeDisplayName, final Date created) {
```

Comment:
Does this work?
```suggestion
    public Event(final UUID id, final String bundleDisplayName, final String applicationDisplayName, final String eventTypeDisplayName, final Timestamp created) {
```

Reply:
Unfortunately it doesn't :disappointed: :

```
org.hibernate.hql.internal.ast.QuerySyntaxException: Unable to locate appropriate constructor on class [com.redhat.cloud.notifications.models.Event]. Expected arguments are: java.util.UUID, java.lang.String, java.lang.String, java.lang.String, java.util.Date [SELECT NEW com.redhat.cloud.notifications.models.Event( e.id, e.bundleDisplayName, e.applicationDisplayName, e.eventTypeDisplayName, e.created) FROM com.redhat.cloud.notifications.models.Event AS e WHERE e.orgId = :orgId AND e.created <= :createdMax]
```

That is why I went with the uglier date-to-instant-to-timestamp conversion...

Reply:
😢